### PR TITLE
DOC: Add example for np.ma.diag

### PR DESCRIPTION
**Pull request by stefmolin**
Add example for np.ma.diag as part of https://github.com/numpy/numpy/issues/22269

### CODE REVIEW

**Comment by melissawm** [comment]
Hi @stefmolin - thanks for the PR! Could you provide some commentary on the example, so it's a little bit less dry? In my experience this makes it richer and more useful. Thanks!

File: numpy/ma/core.py [hunk 1]
```diff
@@ -7096,6 +7096,30 @@ def diag(v, k=0):
     --------
     numpy.diag : Equivalent function for ndarrays.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> x = np.array([[11.2, -3.973, 18], [0.801, -1.41, 12], [7, 33, -12]])
+    >>> mask = [row < 0 for row in x]
+    >>> masked_x = np.ma.masked_array(x, mask)
```
Comment (seberg) [inline numpy/ma/core.py:7104]: Just a note, but this should be simplified (not sure if it is copied from elsewhere).  There is no reason for that list comprehension, and e.g. `masked_where` is maybe more readable also.

**Comment by seberg** [comment]
@stefmolin thanks for the followup, it looks like a good improvement now.  However, the tests are failing: https://dev.azure.com/numpy/numpy/_build/results?buildId=27670&view=logs&jobId=04e3746b-8857-5383-a5f1-d85228b91063&j=04e3746b-8857-5383-a5f1-d85228b91063&t=50651b50-5909-59e8-1ee0-00bfcde49086

It reads like you are using `ma.function` rather than `np.ma.function`, but not sure.

**Review by rossbar — APPROVED** [review]
LGTM, thanks @stefmolin !

**Pull request merged**